bundles/alpha.model/src/alpha/model/transformation/RaiseDependence.xtend

```diff
@@ -77,19 +85,47 @@ import alpha.model.AlphaCompleteVisitable
  *     ((f1 @ E1) op (f2 @ E2)) goes to ((f')@((f1' @ E1) op (f2' @ E2))) where f1 = f' @ f1' and f2 = f' @ f2'
  *     op(f1@E1, f2@E2, ...) goes to (f')@ op(f1'@E1, f2'@E2, ...) where fn = f' @ fn'
  *     case {f1@E1, f2@E2, ...} goes to (f')@ case{f1'@E1, f2'@E2, ...} where fn = f' @ fn'
+ * 
+ * Reduce Expressions:
+ *     The current implementation does not pull dependence expressions out of reduce expressions.
+ *     However, the child of any raised dependence expressions in reduction bodies may be separated
+ *     into a new variable. This is controlled by hoistFromReduce flag in apply. If hoistFromReduce
+ *     is passed as true, then the following rule is applied.
+ * 
+ *     reduce(op, f, g@E) goes to reduce(op, f, g@V) where V is a new local variable defined as V=E
  */
 class RaiseDependence extends AbstractAlphaCompleteVisitor {
+	
+	/** Dependence expressions raised in the body of a reduction may be hoisted into a  
+	 *  separate equation. This flag controls when to do this. See outReduceExpression
+	 *  and reduceExpressionRules.
+	 */
+	protected boolean hoistFromReduce = false
+	
 	/** Protected constructor to restrict access to the instance methods. */
 	protected new() {}
 	
+	protected new(boolean hoistFromReduce) {
+		this()
```

Comment:
With the new constructor that takes in a boolean, we can remove the default constructor as it's not used, and Java won't create the implicit default constructor since a different constructor is declared.

This also lets us declare `hoistFromReduce` as `val` and remove the default value of `false`, that way the value always comes from the constructor.

Reply:
Nice, this is updated.